WordPress/wordpress-playground
# wp-now: accept arguments for `wp-now php` command

## What?


Accept multiple arguments for `wp-now php` command.

## Why?


- Solves https://github.com/WordPress/wordpress-playground/issues/437

## How?



The user can avoid the arguments collision by using the double dash `--`.

## Testing Instructions

- Create a example.php file: `<?php echo 'This is "example.php" file and the php version is: ' . phpversion();`
- Run `npm link`

- The argument `--php=7.4` is shared between wp-now and the php instance:
  - Run `wp-now php example.php --foo=bar --php=7.4` 
  - Observe the output `This is "example.php" file and the php version is: 7.4.31-dev`

- The argument `--php=7.4` is only used by the php instance:
  - Run `wp-now php -- example.php --foo=bar --php=7.4` 
  - Observe the output `This is "example.php" file and the php version is: 8.0.10-dev`

- The argument `--php=7.4` is only passed to the php instance, and the `--php=8.2` is used only by wp-now:
  - Run `wp-now php --php=8.2 -- example.php --foo=bar --php=7.4` 
  - Observe the output `This is "example.php" file and the php version is: 8.2.0-dev`
